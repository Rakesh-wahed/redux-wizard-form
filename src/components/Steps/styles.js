import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const StepsWrapper = styled.div`
  white-space: nowrap;
  overflow: hidden;
`;

export const StepWrapper = styled.div`
  display: inline-block;
  opacity: 0;
  transform: translateX(0);
  transition: opacity 500ms cubic-bezier(0.45, 0.07, 0.83, 0.67), transform 750ms cubic-bezier(0.21, 1, 1, 0.89);
  width: 100%;
`;

export const PointsWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  margin-top: 10px;
`;

export const PointWrapper = styled.span`
  background-color: #9b9393;
  border-radius: 50%;
  height: 10px;
  margin-right: 5px;
  opacity: ${props => (props.isActive ? '1' : '0.5')};
  width: 10px;
`;
