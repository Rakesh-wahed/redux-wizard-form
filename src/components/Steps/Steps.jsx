import React from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import { reduxWizardStep } from '../../containers';
import { Wrapper, StepsWrapper, PointsWrapper, PointWrapper, StepWrapper } from './styles';

export const WizardSteps = props => {
  const stepTransition = {
    entering: {
      opacity: 0,
      transform: `translateX(${(props.currentStep - 1) * -100}%)`
    },
    entered: {
      opacity: 1,
      transform: `translateX(${props.currentStep * -100}%)`
    },
    exited: {
      opacity: 0,
      transform: `translateX(${props.currentStep * -100}%)`
    }
  };

  function renderStep(InnerStep, i) {
    const WizardStep = reduxWizardStep(props.formOptions);

    return (
      <Transition key={i} in={props.currentStep === i} timeout={0}>
        {state => (
          <StepWrapper style={{ ...stepTransition[state] }}>
            <WizardStep Step={InnerStep} {...props} />
          </StepWrapper>
        )}
      </Transition>
    );
  }

  return (
    <Wrapper>
      <StepsWrapper>{props.steps.map(renderStep)}</StepsWrapper>

      <PointsWrapper>
        {props.steps.map((InnerStep, i) => <PointWrapper key={i} isActive={props.activatedSteps.includes(i)} />)}
      </PointsWrapper>
    </Wrapper>
  );
};

WizardSteps.propTypes = {
  steps: PropTypes.arrayOf(PropTypes.func).isRequired,
  activatedSteps: PropTypes.arrayOf(PropTypes.number).isRequired,
  currentStep: PropTypes.number.isRequired,
  formOptions: PropTypes.object.isRequired,
  onPreviousStep: PropTypes.func.isRequired
};
