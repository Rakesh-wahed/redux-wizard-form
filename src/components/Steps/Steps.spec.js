import 'raf/polyfill';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Steps } from './Steps.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('Steps snapshot', () => {
  const component = shallow(
    <Steps steps={[]} activatedSteps={[]} currentStep={0} onGoToStep={() => {}}>
      {() => null}
    </Steps>
  );

  expect(component).toMatchSnapshot();
});
