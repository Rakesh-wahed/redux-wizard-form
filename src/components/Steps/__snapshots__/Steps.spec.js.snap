// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Steps
    activatedSteps={Array []}
    currentStep={0}
    onGoToStep={[Function]}
    steps={Array []}
  >
    [Function]
  </Steps>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
